install_sct

```diff
@@ -631,6 +631,8 @@ if [ "$ACTUAL_PYTHON" != "$EXPECTED_PYTHON"  ]; then
   exit 1
 fi
 
+# Skip pip==21.2 to avoid dependency resolver issue (https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3593)
+python -m pip install -U "pip!=21.2.*"
```

Comment:
Note: I initially tried to specify the version of pip using `conda` (in the `conda create` step, for example), but no matter what I did, `conda` kept thinking that [`pip!=21.2.*` required `python=3.9`](https://github.com/spinalcordtoolbox/spinalcordtoolbox/runs/4254090238?check_suite_focus=true#step:3:334). And, I tried [a _lot_](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/workflows/tests.yml) of different syntax combinations to no avail. 😓  

So, I'm just using `pip` to update itself here instead.